content/tutorials/channel-rewind.textile

```diff
@@ -0,0 +1,226 @@
+  window.addEventListener("load", function () {
+      var ably = new Ably.Realtime("<YOUR-API-KEY>"),
+          $resultWeather = $('#result-open-weather')
+          channelWeather;
+  });
+</script>
+```
+
+Make sure to add in your own Ably API Key in the above code snippet.
+
+We now need to retrieve the user selection from the drop down menu and use the correlated code to subscribe to that particular channel. We'll add the rewind functionality here and set its value to 1, signifying the number of previously published messages we'd like to retrieve from this data stream, on attaching to a channel.
+
+Add the following within your script tag:
+
+```[html]
+subscribeOpenWeather($('select#city-id').val());
+$('select#city-id').on('change', function () {
+    subscribeOpenWeather($('select#city-id').val());
+});
+
+function subscribeOpenWeather(id) {
+    // get the channel object with the rewind parameter set to 1
+    channelWeather = ably.channels.get('[product:ably-openweathermap/weather?rewind=1]weather:' + id);
+    let weatherDesc;
+    // subscribe to this channel and update the text box when new updates arrive
+    channelWeather.subscribe((msg) => {
+        if ($('select#city-id').val() == msg.data.id) {
+            weatherDesc = msg.data.weather[0].description;
+            $resultWeather.text((msg.data.main.temp - 273.15).toFixed(2) + '°C with ' + msg.data.weather[0].description);
+        }
+    });
+}
+```
+
+If you run this file in your browser now, you should see the realtime data coming through from the weather data stream. Well, even if a new update wasn't published just when you run this app, you can still see the most recent update previously sent on this channel, thanks to the channel's rewind param.
+
+"See the full source code on GitHub":https://github.com/ably/tutorials/tree/channel-rewind-ablyjs
+
+h3(#using-sse). Option 2: Consume rewound and new data using Ably's SSE protocol adapter
+
+Let us now look at consuming the same data stream with the rewind parameter, but only this time using Server-Sent Events (SSE) instead of Ably's native client SDKs. Note that since we'll use the SSE protocol, we don't need to use any client SDK at all. We'll make use of the @EvenSource()@ API that comes with all major browsers by default. Replace the script in the previous step with the following:
+
+```[html]
+<script>
+    var $resultWeather = $('#result-open-weather'),
+        oldTemperature = -50,
+        channelWeather,
+        eventSource;
+    var apiKey = "<YOUR-API-KEY>";
+    $('select#city-id').on('change', function () {
+        eventSource.close();
+        subscribeOpenWeather($('select#city-id').val());
+    });
+    subscribeOpenWeather($('select#city-id').val());
+    function subscribeOpenWeather(id) {
+        var channelWeather = "[product:ably-openweathermap/weather?rewind=1]weather:" + id;
+        var URL = `https://realtime.ably.io/sse?v=1.1&key=${apiKey}&channels=${channelWeather}`;
```

Comment:
If you want to include a channel name with params in the name, you need to uri-encode it, which you're not doing here. A much easier way is just to add `rewind=1` as another url param. This is explained in the documentation pr:
https://github.com/ably/docs/pull/729/files#diff-7b572e68f8441588d6eacc180bc8f989R76-R89

Reply:
(to explain this a bit further: the reason this is especially necessary here is that question marks have special meaning in urls, they start the query string ("/sse?v=..."). Maybe in some browsers having subsequent question marks inside the query string will work, but it's not guaranteed to work in all of them. The correct/safe thing to do that will work in all browsers is uri-encode them. (Really this should be done for all api streamer sse examples -- the brackets and forwardslash in `[product:foo/bar]baz` are technically also disallowed, though again, will probably work in some browsers))